msm/src/ffa/interpreter.rs

```diff
@@ -24,15 +25,25 @@ pub trait FFAInterpreterEnv<F: PrimeField> {
 
     fn read_column(&self, ix: FFAColumnIndexer) -> Self::Variable;
 
+    /// Checks |x| = 1, that is x ∈ {-1,0,1}
+    fn range_check_abs1(&mut self, value: &Self::Variable);
+
+    /// Checks input x ∈ [0,2^15)
+    fn range_check_15bit(&mut self, value: &Self::Variable);
+
     /// In constraint environment does nothing (?). In witness environment progresses to the next row.
     fn next_row(&mut self);
 }
 
+fn limb_decompose_bui<F: PrimeField>(input: BigUint) -> [F; N_LIMBS] {
```

Comment:
bui? BigUint?

Reply:
Yes, too C++ style? I could do `limb_decompes_biguint` instead?

Reply:
Changed to `limb_decompose_biguint`